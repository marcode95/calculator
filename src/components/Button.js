import React from 'react';
import PropTypes from 'prop-types';
import './style.css';
import './reset.css';

const Button = ({
  buttonName, clickHandler, wide, color,
}) => (
  <>
    <button
      type="button"
      onClick={() => clickHandler(buttonName)}
      style={{ width: wide ? '50%' : '25%', backgroundColor: color ? 'orange' : '' }}
    >
      {buttonName}
      {' '}
    </button>
  </>
);

Button.propTypes = {
  buttonName: PropTypes.string,
  clickHandler: PropTypes.func.isRequired,
  wide: PropTypes.bool.isRequired,
  color: PropTypes.bool.isRequired,
};

Button.defaultProps = {
  buttonName: 'button',
};

export default Button;
