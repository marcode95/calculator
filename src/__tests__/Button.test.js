import renderer from 'react-test-renderer';
import { render, screen } from '@testing-library/react';
import Button from '../components/Button';
import '@testing-library/jest-dom/extend-expect';

describe('checks if Button renders', () => {
  const clickHandler = () => {
    '';
  };

  test('should render', () => {
    const tree = renderer
      .create(<Button buttonName=""
      wide={false}
      color={false}
      clickHandler={clickHandler} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('should display right value', () => {
    render(<Button buttonName="+"
    wide={true}
    color={false} 
    clickHandler={clickHandler} />);
    const result = screen.getByText('+');

    expect(result).toBeInTheDocument();
  });
});