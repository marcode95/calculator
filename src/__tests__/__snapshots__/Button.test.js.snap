// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`checks if Button renders should render 1`] = `
<button
  onClick={[Function]}
  style={
    Object {
      "backgroundColor": "",
      "width": "25%",
    }
  }
  type="button"
>
  
   
</button>
`;
